src/topology/subset_properties.lean

```diff
@@ -515,40 +525,109 @@ is_connected_of_is_irreducible is_irreducible_empty
+  (H : ∀ x y ∈ s, ∃ t ⊆ s, x ∈ t ∧ y ∈ t ∧ is_connected t) :
+  is_connected s :=
+begin
+  rintros u v hu hv hs ⟨x, xs, xu⟩ ⟨y, ys, yv⟩,
+  rcases H x y xs ys with ⟨t, ts, xt, yt, ht⟩,
+  have := ht u v hu hv(subset.trans ts hs) ⟨x, xt, xu⟩ ⟨y, yt, yv⟩,
+  exact this.imp (λ z hz, ⟨ts hz.1, hz.2⟩)
+end
+
+/-- A union of a family of connected sets with a common point is connected as well. -/
 theorem is_connected_sUnion (x : α) (c : set (set α)) (H1 : ∀ s ∈ c, x ∈ s)
   (H2 : ∀ s ∈ c, is_connected s) : is_connected (⋃₀ c) :=
 begin
-  rintro u v hu hv hUcuv ⟨y, hyUc, hyu⟩ ⟨z, hzUc, hzv⟩,
-  cases classical.em (c = ∅) with hc hc,
-  { rw [hc, sUnion_empty] at hyUc, exact hyUc.elim },
-  cases ne_empty_iff_exists_mem.1 hc with s hs,
-  cases hUcuv (mem_sUnion_of_mem (H1 s hs) hs) with hxu hxv,
-  { rcases hzUc with ⟨t, htc, hzt⟩,
-    specialize H2 t htc u v hu hv (subset.trans (subset_sUnion_of_mem htc) hUcuv),
-    cases H2 ⟨x, H1 t htc, hxu⟩ ⟨z, hzt, hzv⟩ with r hr,
-    exact ⟨r, mem_sUnion_of_mem hr.1 htc, hr.2⟩ },
-  { rcases hyUc with ⟨t, htc, hyt⟩,
-    specialize H2 t htc u v hu hv (subset.trans (subset_sUnion_of_mem htc) hUcuv),
-    cases H2 ⟨y, hyt, hyu⟩ ⟨x, H1 t htc, hxv⟩ with r hr,
-    exact ⟨r, mem_sUnion_of_mem hr.1 htc, hr.2⟩ }
+  apply is_connected_of_forall x,
+  rintros y ⟨s, sc, ys⟩,
+  exact ⟨s, subset_sUnion_of_mem sc, H1 s sc, ys, H2 s sc⟩
 end
 
-theorem is_connected_union (x : α) {s t : set α} (H1 : x ∈ s) (H2 : x ∈ t)
+theorem is_connected.union (x : α) {s t : set α} (H1 : x ∈ s) (H2 : x ∈ t)
   (H3 : is_connected s) (H4 : is_connected t) : is_connected (s ∪ t) :=
-have _ := is_connected_sUnion x {t,s}
-  (by rintro r (rfl | rfl | h); [exact H1, exact H2, exact h.elim])
-  (by rintro r (rfl | rfl | h); [exact H3, exact H4, exact h.elim]),
-have h2 : ⋃₀ {t, s} = s ∪ t, from (sUnion_insert _ _).trans (by rw sUnion_singleton),
-by rwa h2 at this
+sUnion_pair s t ▸ is_connected_sUnion x {s, t}
+  (by rintro r (rfl | rfl | h); [exact H2, exact H1, exact h.elim])
+  (by rintro r (rfl | rfl | h); [exact H4, exact H3, exact h.elim])
 
-theorem is_connected_closure {s : set α} (H : is_connected s) :
+theorem is_connected.closure {s : set α} (H : is_connected s) :
   is_connected (closure s) :=
 λ u v hu hv hcsuv ⟨y, hycs, hyu⟩ ⟨z, hzcs, hzv⟩,
 let ⟨p, hpu, hps⟩ := exists_mem_of_ne_empty (mem_closure_iff.1 hycs u hu hyu) in
 let ⟨q, hqv, hqs⟩ := exists_mem_of_ne_empty (mem_closure_iff.1 hzcs v hv hzv) in
 let ⟨r, hrs, hruv⟩ := H u v hu hv (subset.trans subset_closure hcsuv) ⟨p, hps, hpu⟩ ⟨q, hqs, hqv⟩ in
 ⟨r, subset_closure hrs, hruv⟩
 
+theorem is_connected.image [topological_space β] {s : set α} (H : is_connected s)
+  (f : α → β) (hf : continuous_on f s) : is_connected (f '' s) :=
+begin
+  -- Unfold/destruct definitions in hypotheses
+  rintros u v hu hv huv ⟨_, ⟨x, xs, rfl⟩, xu⟩ ⟨_, ⟨y, ys, rfl⟩, yv⟩,
+  rcases continuous_on_iff'.1 hf u hu with ⟨u', hu', u'_eq⟩,
+  rcases continuous_on_iff'.1 hf v hv with ⟨v', hv', v'_eq⟩,
+
```

Comment:
Can you remove the empty line, here and below?

Reply:
Done.